packages/battery_plus/battery_plus/example/integration_test/battery_plus_test.dart

```diff
@@ -23,6 +35,26 @@ void main() {
     final battery = Battery();
     final isInBatterySaveMode = await battery.isInBatterySaveMode;
     debugPrint('$isInBatterySaveMode');
-    expect(isInBatterySaveMode, isNotNull);
+    expect(isInBatterySaveMode, isFalse);
+  });
+
+  testWidgets('test if device is in low power mode',
+      (WidgetTester tester) async {
+    await Process.run(
+        'adb', ['shell', 'dumpsys', 'battery', 'set', 'level', '5'],
+        runInShell: true);
+//    await Process.run('adb', [
+//      'shell',
+//      'pm',
+//      'grant',
+//      'dev.steenbakker.mobile_scanner_example',
+//      'android.permission.CAMERA'
+//    ]);
+    final battery = Battery();
+    final isInBatterySaveMode = await battery.isInBatterySaveMode;
+    if (isInBatterySaveMode) {
+      debugPrint('$isInBatterySaveMode');
+      expect(isInBatterySaveMode, isTrue);
+    }
```

Comment:
FYI, You shouldn't have these type of conditional flow inside tests.

If `isInBatterySaveMode` is false, the test will pass because the `expect` condition is never run.

If `isInBatterySaveMode1 is true, the test will pass as well.

I am not sure if this is just WIP but this is plain wrong.